internal/otelcollector/config/trace/gateway/processors.go

```diff
@@ -82,18 +79,63 @@ func makeProcessorsConfig() Processors {
 	}
 }
 
+const (
+	methodIsGet                        = "attributes[\"http.method\"] == \"GET\""
+	methodIsPost                       = "attributes[\"http.method\"] == \"POST\""
+	componentIsProxy                   = "attributes[\"component\"] == \"proxy\""
+	operationIsIngress                 = "(attributes[\"OperationName\"] == \"Ingress\" or IsMatch(name, \"ingress.*\") == true)"
+	operationIsEgress                  = "(attributes[\"OperationName\"] == \"Egress\" or IsMatch(name, \"egress.*\") == true)"
+	namespacesIsKymaSystem             = "resource.attributes[\"k8s.namespace.name\"] == \"kyma-system\""
+	namespacesIsIstioSystem            = "resource.attributes[\"k8s.namespace.name\"] == \"istio-system\""
+	istioNameIsGrafana                 = "attributes[\"istio.canonical_service\"] == \"grafana\""
+	istioNameIsAuthProxy               = "attributes[\"istio.canonical_service\"] == \"monitoring-auth-proxy-grafana\""
+	istioNameIsFluentBit               = "attributes[\"istio.canonical_service\"] == \"telemetry-fluent-bit\""
+	istioNameIsTraceGateway            = "attributes[\"istio.canonical_service\"] == \"telemetry-trace-collector\""
+	istioNameIsMetricAgent             = "attributes[\"istio.canonical_service\"] == \"telemetry-metric-agent\""
+	istioNameIsMetricGateway           = "attributes[\"istio.canonical_service\"] == \"telemetry-metric-gateway\""
+	istioNameIsIstioGateway            = "attributes[\"istio.canonical_service\"] == \"istio-ingressgateway\""
+	userAgentIsVMScrapeAgent           = "attributes[\"user_agent\"] == \"vm_promscrape\""
+	userAgentIsPrometheus              = "IsMatch(attributes[\"user_agent\"], \"Prometheus\\\\/.*\") == true"
+	userAgentIsKymaOtelCol             = "IsMatch(attributes[\"user_agent\"], \"kyma-otelcol\\\\/.*\") == true"
+	urlIsIstioHealthz                  = "IsMatch(attributes[\"http.url\"], \"https:\\\\/\\\\/healthz\\\\..+\\\\/healthz\\\\/ready\") == true"
+	urlIsTelemetryTraceService         = "IsMatch(attributes[\"http.url\"], \"http(s)?:\\\\/\\\\/telemetry-otlp-traces\\\\.kyma-system(\\\\..*)?:(4317|4318).*\") == true"
+	urlIsTelemetryTraceInternalService = "IsMatch(attributes[\"http.url\"], \"http(s)?:\\\\/\\\\/telemetry-trace-collector-internal\\\\.kyma-system(\\\\..*)?:(55678).*\") == true"
+	urlIsTelemetryMetricService        = "IsMatch(attributes[\"http.url\"], \"http(s)?:\\\\/\\\\/telemetry-otlp-metrics\\\\.kyma-system(\\\\..*)?:(4317|4318).*\") == true"
+	and                                = " and "
+
+	toFromKymaGrafana            = componentIsProxy + and + namespacesIsKymaSystem + and + istioNameIsGrafana
```

Comment:
Let's further simplify it by adding the following functions:
```go
func spanAttributeEquals(key, value string) string {
	return "attributes[\"" + key + "\"] == \"" + value + "\""
}

func resourceAttributeEquals(key, value string) string {
	return "resource.attributes[\"" + key + "\"] == \"" + value + "\""
}

func urlMatches(pattern string) string {
	return "IsMatch(attributes[\"http.url\"], \"" + pattern + "\") == true"
}

func userAgentMatches(pattern string) string {
	return "IsMatch(attributes[\"user_agent\"], \"" + pattern + "\") == true"
}

func joinWithAnd(parts ...string) string {
	return strings.Join(parts, " and ")
}
```

Reply:
done